### ISSUE TITLE
 [$500] Feature Request: Add :emojicode: selectors when editing messages.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#34442]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Problem:
Add :emojicode: selectors when editing messages.

## Solution: 

As a NewDot user I have become accustomed to being able to express myself with emojis, and the default way I do that is by typing :emojicodes:. However, when editing an existing message, we have an emoji picker, but no support for :emojicodes:

## Context/Examples/Screenshots/Notes:

**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @roryabraham 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1704991655069279

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b0abd6fa6e877b41</li>
        <li>Upwork Job ID: 1745886924480851968</li>
        <li>Last Price Increase: 2024-01-19</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 28120470</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

**Comment by laurenreidexpensify** [comment]
@eVoloshchak bump on review thanks

**Comment by eVoloshchak** [comment]
I think we should proceed with @neonbhai's [proposal](https://github.com/Expensify/App/issues/34442#issuecomment-1889426112)
Using `ComposerWithSuggestions` is the cleaner approach, it will allow us to reuse an already existing logic 
I do agree with the following concern from @dukenv0307 though
> We shouldn't implement ComposerWithSuggestions for ReportActionItemMessageEdit because the global composer has many things different logic.

But I don't think creating a third component with overlapping logic would be a better solution. `ComposerWithSuggestions` ideally should just be a `Composer` with `Suggestions` component (that is not entirely the case now, the logic in `ComposerWithSuggestions` differs a little, but introducing a prop to control it in case we need to would be simpler)

🎀👀🎀 C+ reviewed!

**Comment by laurenreidexpensify** [comment]
next step: @stitesExpensify to assign

**Comment by stitesExpensify** [comment]
@eVoloshchak does this change your opinion on which approach we should take? https://github.com/Expensify/App/issues/34442#issuecomment-1899049614

**Comment by eVoloshchak** [comment]
> does this change your opinion on which approach we should take? https://github.com/Expensify/App/issues/34442#issuecomment-1899049614

@stitesExpensify, it does. I took a look at @dukenv0307's test branch and `ComposerWithSuggestions` once again and I agree, there is too much logic that is specific to global composer, it would be simpler to just have a separate component. We should think of a better name for `ComposerWithSuggestionsEdit` on the PR stage

Let's proceed with @dukenv0307, sorry for the confusion folks

**Comment by eVoloshchak** [comment]
Not overdue, PR is being drafted in https://github.com/Expensify/App/pull/35226

**Comment by laurenreidexpensify** [comment]
Not overdue, PR is being drafted in https://github.com/Expensify/App/pull/35226

**Comment by laurenreidexpensify** [comment]
Not overdue

**Comment by eVoloshchak** [comment]
Not overdue

**Comment by laurenreidexpensify** [comment]
PR is still in draft

**Comment by laurenreidexpensify** [comment]
@dukenv0307 how's the PR going?

**Comment by laurenreidexpensify** [comment]
PR is in active review

**Comment by laurenreidexpensify** [comment]
PR still in review

**Comment by laurenreidexpensify** [comment]
PR is still in active review

**Comment by laurenreidexpensify** [comment]
Looks like we're super close to merging 🤞

**Comment by laurenreidexpensify** [comment]
@dukenv0307 can you confirm latest on this one? Thanks

**Comment by joekaufmanexpensify** [comment]
@dukenv0307 , you were assigned to https://github.com/Expensify/App/issues/40864#issuecomment-2084529572 and the PR from this issue is fixing the case from the issue I linked there. We're going to close out the other one, so we don't need to track it in multiple places. 

Let us know if you have any Q's.

**Comment by laurenreidexpensify** [comment]
Still not ready for payment, PR still in draft

**Comment by stitesExpensify** [comment]
Still in draft

**Comment by laurenreidexpensify** [comment]
Still not ready for payment, PR still in draft

**Comment by laurenreidexpensify** [comment]
@dukenv0307 can you confirm PR is still in draft thanks

**Comment by laurenreidexpensify** [comment]
still in draft

**Comment by laurenreidexpensify** [comment]
@dukenv0307 @stitesExpensify we still in draft?

**Comment by stitesExpensify** [comment]
Looks like there was a C+ review last week and we are waiting on @dukenv0307 to update some things

**Comment by laurenreidexpensify** [comment]
@dukenv0307 @stitesExpensify any update here? thanks

**Comment by eVoloshchak** [comment]
@laurenreidexpensify, https://github.com/Expensify/App/pull/41071 is still being worked on, we're facing an [error](https://github.com/Expensify/App/pull/41071#pullrequestreview-2110331763)

**Comment by laurenreidexpensify** [comment]
@twisterdotcom handing this one over for parental leave cover. We're still in review for the PR, so this will just be handling payment. We're also blocked on https://github.com/Expensify/App/pull/41071 I believe. Thanks!

**Comment by eVoloshchak** [comment]
This feature was added by another PR (https://github.com/Expensify/App/pull/44720) already it turns out
There is one small UI issue (https://github.com/Expensify/App/pull/41071#pullrequestreview-2195746266), but otherwise, this is fully implemented 

@laurenreidexpensify, @stitesExpensify, what would be the correct course of action here?
Should we at least fix this UI issue?

**Comment by twisterdotcom** [comment]
We should fix the UI issue and pay out an acceptable amount for the work done up until now. I don't have a good sense of it, but I would say $500 might not be it. @eVoloshchak what's your honest assessment?

**Comment by eVoloshchak** [comment]
This is a second PR (the first one contained a regression and had to be reverted), so the full payment would have been $250. Given that and the fact that the UI fix would be straightforward (just need to change a single padding number), $150 seems fair to me

**Comment by twisterdotcom** [comment]
$150 it is. Payment Summary:

- @eVoloshchak paid $150 via NewDot requests (C+)
- @dukenv0307 paid $150 via Upwork [here](https://www.upwork.com/nx/wm/offer/103495012) (Contributor)

**Comment by twisterdotcom** [comment]
I already did all the payment summary here.

**Comment by JmillsExpensify** [comment]
$150 approved for @eVoloshchak
